src/arch-clap.ts

```diff
@@ -0,0 +1,518 @@
+      task: ["rm-eslint-deps"]
+    },
+    "test-only": {
+      desc: "Run just your unit tests (generate test/mocha.opts if not exist)",
+      task: () => {
+        xarcModuleDev.setupPublishingConfig();
+        xarcModuleDev.setupMocha();
+        return ".test-only";
+      }
+    },
+    ".test-only": `mocha -c test/spec`,
+    ".test-cov": `nyc clap -q test-only`,
+    "test-cov": {
+      desc: "Use nyc to generate coverage for tests (add nyc config to your package.json)",
+      task: () => {
+        xarcModuleDev.setupCoverage();
+        return ".test-cov";
+      }
+    },
+    "add-typescript-deps": {
+      desc: "Add dependencies for typescript support to your package.json",
+      task: () => xarcModuleDev.addDevDeps("typescript")
+    },
+    "add-eslint-deps": {
+      desc: "Add dependencies for eslint support to your package.json",
+      task: () => xarcModuleDev.addDevDeps("eslint")
+    },
+    "rm-typescript-deps": {
+      desc: "Remove dependencies for typescript support from your package.json",
+      task: () => xarcModuleDev.rmDevDeps("typescript")
+    },
+    "rm-eslint-deps": {
+      desc: "Remove dependencies for eslint support from your package.json",
+      task: () => xarcModuleDev.rmDevDeps("eslint")
+    }
+  };
+
+  /* if linting enable, then add eslint tasks */
+  if (lint) {
+    let eslintDir = Path.normalize(`${__dirname}/../config/eslint`);
+
+    const customDir = Path.resolve("eslint");
+    if (Fs.existsSync(customDir)) {
+      eslintDir = customDir;
+    }
+
+    const lintTasks = {
+      ".lint-src-ts": `eslint -c ${eslintDir}/.eslintrc-node-ts src --ext .ts,.tsx --color`,
+      ".lint-src-js": `eslint -c ${eslintDir}/.eslintrc-node src --ext .js,.jsx --color`,
+      ".lint-lib-ts": `eslint -c ${eslintDir}/.eslintrc-node-ts lib --ext .ts,.tsx --color`,
+      ".lint-lib-js": `eslint -c ${eslintDir}/.eslintrc-node lib --ext .js,.jsx --color`,
+      ".lint-test-ts": `eslint -c ${eslintDir}/.eslintrc-test-ts test/spec --ext .ts,.tsx --color`,
+      ".lint-test-js": `eslint -c ${eslintDir}/.eslintrc-test test/spec --ext .js,.jsx --color`,
+      lint: [invokeLint()]
+    };
+
+    Object.assign(tasks, lintTasks);
+  } else {
+    Object.assign(tasks, { lint: "echo linting is disabled by option enableLinting set to false" });
+  }
+
+  return tasks;
+}
+
+/**
+ * Load XClap build tasks for developing node.js modules.
+ *
+ * See document for xclap at https://www.npmjs.com/package/xclap
+ *
+ * To use, create a file `xclap.js`:
+ *
+ * ```js
+ *   require("@xarc/module-dev")()
+ * ```
+ *
+ * or `xclap.ts` for typescript:
+ *
+ * ```ts
+ *   import loadTasks from "@xarc/module-dev";
+ *   loadTasks();
+ * ```
+ *
+ * Then run the command `npx clap` to see available build tasks.
+ *
+ * @param xclapOrOptions options
+ */
+export function loadTasks(xclapOrOptions?: object | XarcModuleDevOptions) {
+  let options = {} as XarcModuleDevOptions;
```

Comment:
I would highly suggest finding a way to get rid of these casts. This is effectively lying to TS and will cause it to not be able to find bugs. You could do something like this:

```ts
function buildOptions(xclapOrOptions?: XarcModuleDevOptions): XarcModuleDevOptions {
  if (!xclapOrOptions) {
    return {};
  }

  const cname = xclapOrOptions.constructor.name;
  if (cname !== "XClap") {
    return options;
  }

  return { xclap: xclapOrOptions };
}

export function loadTasks(xclapOrOptions?: object | XarcModuleDevOptions) {
  const options = buildOptions(xclapOrOptions);
  // ...
}
```